applications/CompressiblePotentialFlowApplication/python_scripts/apply_far_field_process.py

```diff
@@ -15,64 +20,68 @@ def __init__(self, Model, settings ):
-
-        #select the first node
-        for node in self.model_part.Nodes:
-            node1 = node
+        #KratosMultiphysics.VariableUtils().SetVectorVar(CPFApp.VELOCITY_INFINITY, self.velocity_infinity, self.far_field_model_part.Conditions)
+        for cond in self.far_field_model_part.Conditions:
+            cond.SetValue(CPFApp.VELOCITY_INFINITY, self.velocity_infinity)
+
+        # Select the first node in the mesh as reference
+        for node in self.far_field_model_part.Nodes:
+            x0 = node.X
+            y0 = node.Y
+            z0 = node.Z
             break
 
-        #find the node with the minimal x
-        x0 = node1.X
-        y0 = node1.X
-        z0 = node1.X
-
+        # Find smallest distance_to_reference
         pos = 1e30
-        for node in self.model_part.Nodes:
-            dx = node.X - x0
-            dy = node.Y - y0
-            dz = node.Z - z0
-
-            tmp = dx*self.velocity_infinity[0] + dy*self.velocity_infinity[1] + dz*self.velocity_infinity[2]
-
-            if(tmp < pos):
-                pos = tmp
-
-        for node in self.model_part.Nodes:
+        for node in self.far_field_model_part.Nodes:
+            # Computing distance to reference
             dx = node.X - x0
             dy = node.Y - y0
             dz = node.Z - z0
 
-            tmp = dx*self.velocity_infinity[0] + dy*self.velocity_infinity[1] + dz*self.velocity_infinity[2]
-
-            if(tmp < pos+1e-9):
-                node.Fix(CompressiblePotentialFlowApplication.VELOCITY_POTENTIAL)
-                node.SetSolutionStepValue(CompressiblePotentialFlowApplication.VELOCITY_POTENTIAL,0,self.inlet_phi)
-                if self.model_part.HasNodalSolutionStepVariable(CompressiblePotentialFlowApplication.ADJOINT_VELOCITY_POTENTIAL):
-                    node.Fix(CompressiblePotentialFlowApplication.ADJOINT_VELOCITY_POTENTIAL)
-                    node.SetSolutionStepValue(CompressiblePotentialFlowApplication.ADJOINT_VELOCITY_POTENTIAL,0,0.0)
+            distance_to_reference = dx*self.velocity_infinity[0] + dy*self.velocity_infinity[1] + dz*self.velocity_infinity[2]
+
+            if(distance_to_reference < pos):
+                pos = distance_to_reference
+                self.reference_inlet_node = node
+
+        # Fix nodes in the inlet
+        for cond in self.far_field_model_part.Conditions:
+            normal = cond.GetValue(KratosMultiphysics.NORMAL)
+            projection = DotProduct(normal,self.velocity_infinity)
+
+            if( projection < 0):
+                for node in cond.GetNodes():
+                    # Computing distance to reference
+                    dx = node.X - self.reference_inlet_node.X
+                    dy = node.Y - self.reference_inlet_node.Y
+                    dz = node.Z - self.reference_inlet_node.Z
+
+                    inlet_phi = dx*self.velocity_infinity[0] + dy*self.velocity_infinity[1] + dz*self.velocity_infinity[2]
```

Comment:
Should `self.velocity_infinity` be a unit vector? If so, please check this in the constructor. Also, consider the choice of name for easier understanding. E.g., `free_stream_direction`.

Reply:
self.velocity_infinity is a vector equal to the magnitude to the velocity of the undisturbed free stream far away from the body. It is therefore not a unit vector. I changed the name to free_stream_velocity as you suggested in https://github.com/KratosMultiphysics/Kratos/pull/4668 to make it more clear.